doc/source/actions.rst

```diff
@@ -351,6 +351,26 @@ The ``merge`` action merges the pull request into its base branch. The
        Note that the ``rebase`` method has many drawbacks, see :ref:`strict
        rebase`.
 
+   * - ``strict_smart_behavior``
+     - string
+     - ``merge-if-ready``
+     - Behavior when a pull request is ready to be merge in ``strict=smart``
+       mode. Possible values:
+
+       * ``merge-if-ready`` if the pull request is ready and already in sync
+         with its base branch, the queue is by-passed and the pull request is
+         directly merged. This behavior allows to safe CIs resources.
+       * ``always-append`` pull requests are always placed at the end of the queue
+
+   * - ``strict_smart_priority_labels``
+     - list of string
+     - `[]`
+     - List of labels to prioritize pull request in queue. Labels must be
+       ordered from the highest priority to the lowest.
+
+       Once a pull request enters in the queue, the priority can be changed even
+       if the label is removed.
```

Comment:
I'm wondering if we shouldn't do something different, which is by using `conditions`. Labels are good but they are kind limited.

For example you might want PR from a bot to be always low priority. You could do:

```yaml
- name: automatic merge from dependabot
    conditions:
      - author~=^dependabot(|-preview)\[bot\]$
      - "status-success=ci/circleci: py37"
    actions:
      merge:
        strict: "smart"
        priority: -10
        method: rebase
  - name: automatic merge
    conditions:
      - base=master
      - "status-success=ci/circleci: py38"
      - "#approved-reviews-by>=1"
    actions:
      merge:
        strict: "smart"
        method: rebase
        priority: 10
```

I don't like priority being numbered because you never know which is low which is high.
So I'd say make them an `Integer` where the higher the more priority, so it's flexible, plus add some conveninent alias such as:
- `high` = 128
- `normal` = 0
- ` low` = -128

It makes it easy to understand when writing/reading the configuration.

Reply:
I have thought of this solution too (I have already a branch with that), I prefer it for it's flexibility, but I didn't take this road because I am pretty sure the engine has some race when you have many merge rules. Currently people have no chance to run into them. But if they start switching labels to change the merge rule, this will increase a lot the change to trigger them.

That's said, while I'm writing this, I found a simple idea to fix up the race I have in mind.

I will split this PR to merge the first part and propose the priority thing into another one.

Reply:
Awesome <3